dataprofiler/tests/profilers/helpers/test_report_helpers.py

```diff
@@ -27,5 +28,157 @@ def test_serializable_report(self):
 
+    def test_omit_keys_in_report(self):
+
+        report = {
+            "test0": 0,
+            "test1": 1,
+            "test2": {
+                "test3": 3
+            },
+            "test4": {
+                "test5": {
+                    "test3": 3
+                }
+            }
+        }
+        output_format = 'pretty'
+        omit_keys = []
+
+        prepared_report = _prepare_report(report, output_format, omit_keys)
+        self.assertDictEqual(prepared_report, report)
+
+        omit_keys = ['test0']
+        report_test1 = {
+            "test1": 1,
+            "test2": {
+                "test3": 3
+            },
+            "test4": {
+                "test5": {
+                    "test3": 3
+                }
+            }
+        }
+        prepared_report = _prepare_report(report, output_format, omit_keys)
+        self.assertDictEqual(prepared_report, report_test1)
+
+
+        omit_keys = ['test0', 'test4.test5.test3']
+        report_test2 = {
+            "test1": 1,
+            "test2": {
+                "test3": 3
+            },
+            "test4": {
+                "test5": {
+                }
+            }
+        }
+        prepared_report = _prepare_report(report, output_format, omit_keys)
+        self.assertDictEqual(prepared_report, report_test2)
+
+        
+        omit_keys = ['test0', 'test4.test5']
+        report_test3 = {
+            "test1": 1,
+            "test2": {
+                "test3": 3
+            },
+            "test4": {
+            }
+        }
+        prepared_report = _prepare_report(report, output_format, omit_keys)
+        self.assertDictEqual(prepared_report, report_test3)
+
+        omit_keys = ['test1', 'test4']
+        report_test4 = {
+            "test0": 0,
+            "test2": {
+                "test3": 3
+            }
+        }
+        prepared_report = _prepare_report(report, output_format, omit_keys)
+        self.assertDictEqual(prepared_report, report_test4)
+
+        omit_keys = ['test0', 'test2.test3']
+        report_test5 = {
+            "test1": 1,
+            "test2": {
+            },
+            "test4": {
+                "test5": {
+                    "test3": 3
+                }
+            }
+        }
+        prepared_report = _prepare_report(report, output_format, omit_keys)
+        self.assertDictEqual(prepared_report, report_test5)
+
+        # Keys that don't exist in the report
+        omit_keys = ['test6', 'test3']
+        prepared_report = _prepare_report(report, output_format, omit_keys)
+        self.assertDictEqual(prepared_report, report)
+
+        # Robustness check
+        omit_keys = None
+        prepared_report = _prepare_report(report, output_format, omit_keys)
+        self.assertDictEqual(prepared_report, report)
+
+
+        # Test wildcard removals
+        wild_report1 = {
+            "test0": 0,
+            "test1": 1,
+            "test2": {
```

Comment:
should we remove test2 in this case, as it's an empty dict now.

Reply:
I could probably recurse and do that, I was thinking I'd leave it unless otherwise specified.